pkg/mdformatter/mdformatter.go

```diff
@@ -109,11 +109,21 @@ func (FormatFrontMatter) TransformFrontMatter(_ SourceContext, frontMatter map[s
 	}
 	sort.Sort(sort.Reverse(sort.StringSlice(keys)))
 
-	b := bytes.NewBuffer([]byte("---\n"))
+	b := bytes.NewBuffer([]byte("---"))
 	for _, k := range keys {
-		_, _ = fmt.Fprintf(b, "%v: %v\n", k, frontMatter[k])
+		// Check if frontMatter[k] is a map.
```

Comment:
For the next time, this comment might be not needed - the code is now nicely self-explanatory (: